File: RedSheet.DbEntities/Models/LiteTheirTeamMember.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("LiteTheirTeamMembers", Schema = "dbo")]
    [Serializable]
    public partial class LiteTheirTeamMember
    {

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int LiteTheirTeamMemberId { get; set; }

        [Required]
        [MaxLength(50)]
        public string LiteTheirTeamMemberName { get; set; }

        [Range(1, int.MaxValue)]
        public int PersonalityId { get; set; }

        [Required]
        [MaxLength(50)]
        public string Position { get; set; }

        [RelationshipTableAttribue("LiteProjectBackgrounds", "dbo")]
        //Relationships
        public int LiteProjectBackgroundId { get; set; }

        [ForeignKey("LiteProjectBackgroundId")]
        public LiteProjectBackground LiteProjectBackground { get; set; }

        public LiteTheirTeamMember()
        {
        }

    }
}

File: redsheet-portal/src/app/database-models/lite-their-team-member.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { LiteProjectBackground,  vLiteTheirTeamMemberRecord  } from './'
export class LiteTheirTeamMember {
    constructor(liteTheirTeamMember?: LiteTheirTeamMember  | vLiteTheirTeamMemberRecord )  {
        let properties = [ "liteTheirTeamMemberId", "liteTheirTeamMemberName", "personalityId", "position", "liteProjectBackgroundId",];
        for (let property of properties)
            if (liteTheirTeamMember && liteTheirTeamMember[property])
                this[property] = liteTheirTeamMember[property];
    }

	liteTheirTeamMemberId : number =   0 ;

    @required()
    @maxLength(50)
	liteTheirTeamMemberName : string =   undefined;

    @range(1,2147483647)
	personalityId : number =   undefined;

    @required()
    @maxLength(50)
	position : string =   undefined;

    @range(0,2147483647)
	liteProjectBackgroundId : number =   undefined;
	liteProjectBackground : LiteProjectBackground  ;


}
